### PR TITLE
Looker embed

### PR DESCRIPTION
**Pull request by ps-jspitler**

### CODE REVIEW

File: src/panel/TemplateEditor/DataPanel/PortalView.tsx [hunk 1]
```diff
@@ -15,14 +16,31 @@ export default () => {
         template: templateFile,
         url: inspectedWindowUrl
     }).toString();
-    const url = `${portal?.views?.panel}?${params}`;
-    const seller_id = !!template?.obj?.params?.seller_id ? String(template?.obj?.params?.seller_id) : '';
-    const tempUrl = `${portal?.views?.panel}&hide_filter=Retailer+ID,Seller+ID`.replace(/seller_id/g, seller_id);
+    const [ embedUrl, setEmbedUrl ] = useState<string>();
+
+    useEffect(() => {
```
Comment (dtempx) [inline src/panel/TemplateEditor/DataPanel/PortalView.tsx:21]: Move this into a new PortalContext within the TemplateEditor context such that `panelUrl` comes from `usePortal`.

File: src/panel/TemplateEditor/Sidebar/PortalDialog.tsx [hunk 2]
```diff
@@ -31,7 +32,24 @@ export default ({ open, onClose }: Props) => {
         template: templateFile,
         url: inspectedWindowUrl
     }).toString();
-    const url = `${portal?.views?.full}?${params}`;
+    const [ embedUrl, setEmbedUrl ] = useState<string>();
+    
+    useEffect(() => {
```
Comment (dtempx) [inline src/panel/TemplateEditor/Sidebar/PortalDialog.tsx:37]: Move this into a new PortalContext within the TemplateEditor context such that `dialogUrl` comes from `usePortal`.

File: src/panel/Portal.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
 export interface Portal {
+    looks?: {
```
Comment (dtempx) [inline src/panel/Portal.ts:2]: Just use `views`, no need for back compat.

File: src/panel/TemplateEditor/lib/cloud/looker.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+import * as request from "./request";
+
+export const load = async (lookId: number, filters = {} as { [ key: string ]: unknown }) => {
+    const { url } = await request.json(`/pricespider/looker-embed-url?look=${lookId}&filters=${JSON.stringify(filters)}`);
```
Comment (dtempx) [inline src/panel/TemplateEditor/lib/cloud/looker.ts:5]: Make this distinction about looker and pricespider in the cloud, passing the template path and the template params, such that the editor has no knowledge of pricespider or looker. The user context is already defined in the auth, so if you need to know something about for the pricespider you can get that by querying the users collection on the server.

**Review by dtempx — CHANGES_REQUESTED** [review]

File: src/panel/TemplateEditor/lib/cloud/portal.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+
+import * as request from "./request";
+
+interface PortalParams {
+    params?: Record<string, unknown>;
+}
+
+export const load = async (params: PortalParams, userId?: string) => {
```
Comment (ps-jspitler) [inline src/panel/TemplateEditor/lib/cloud/portal.ts:8]: Should we move this, and `getUser` and other 'new' network calls into syphonx-lib?

File: src/panel/TemplateEditor/lib/cloud/user.ts [hunk 6]
```diff
@@ -2,16 +2,37 @@
...
     getFirestore,
     onSnapshot,
-    doc
+    doc,
+    Timestamp
 } from "firebase/firestore";
 import { initializeApp } from "firebase/app";
 import * as request from "./request";
 
+function formatDocument(obj: unknown): unknown {
```
Comment (ps-jspitler) [inline src/panel/TemplateEditor/lib/cloud/user.ts:11]: the cause of some pretty small / hard-to-diagnose issues... the validation of the user... we might move that 100% to servers-side, to reduce this redundancy?

**Review by dtempx — APPROVED** [review]

**Pull request merged**